scripts/controllers/RobotController.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+"""
+This code serves as the generic RobotController
+class.
+"""
+
+import rospy
+import numpy as np
+from std_msgs.msg import Float64
+from sensor_msgs.msg import JointState
+from enum import Enum
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+from RobotControllerManager import RobotControllerManager
+
+
+class ControllerType(Enum):
+    POSITION = 1
+    VELOCITY = 2
+    TRAJECTORY = 3
+
+
+class RobotArm(object):
+    """
+    Generic robot arm implementation in ROS
+    """
+    def __init__(self, num_joints,
+                 is_sim=True, joint_states_topic='/joint_states',
+                 ignore_joints_arr=['panda_finger_joint1', 'panda_finger_joint2'],
+                 controller_names=None):
+        """
+        controller_names should have -
+        controller_names[0] - position controllers
+        controller_names[1] - velocity controllers
+        controllers_names[2] - ONE trajectory controller.
+        """
+        if controller_names is None:
+            controller_names = []
+            default_pos_names = ['panda_joint%s_position_controller' % (i) for i in range(1, num_joints+1)]
+            default_vel_names = ['panda_joint%s_velocity_controller' % (i) for i in range(1, num_joints+1)]
+            default_traj_name = ['/panda_joint_trajectory_controller/command' if is_sim else '/joint_trajectory_controller/command']
```

Comment:
Are you gonna leave it as `panda_*_controller`?

Reply:
These are the default options currently.